### PR TITLE
Add stats and explicit startup checks for resource groups

### PR DESCRIPTION
**Pull request by raghavsethi**

### CODE REVIEW

File: presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java [hunk 1]
```diff
@@ -302,4 +313,16 @@ private ResourceGroup getRootGroup(ResourceGroupId groupId)
         // GroupId is guaranteed to be in groups: it is added before the first call to this method in configure()
         return groups.get(groupId);
     }
+
+    @Managed
```
Comment (nezihyigitbasi) [inline presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java]: Can you move adding stats into a separate commit too?

File: presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java [hunk 2]
```diff
@@ -144,42 +148,49 @@ public void configure(ResourceGroup group, SelectionContext context)
...
     {
         try {
-            Map.Entry<ManagerSpec, Map<ResourceGroupIdTemplate, ResourceGroupSpec>> specsFromDb = buildSpecsFromDb();
-            ManagerSpec managerSpec = specsFromDb.getKey();
-            Map<ResourceGroupIdTemplate, ResourceGroupSpec> resourceGroupSpecs = specsFromDb.getValue();
-            Set<ResourceGroupIdTemplate> changedSpecs = new HashSet<>();
-            Set<ResourceGroupIdTemplate> deletedSpecs = Sets.difference(this.resourceGroupSpecs.keySet(), resourceGroupSpecs.keySet());
+            load();
+        }
+        catch (Throwable e) {
```
Comment (nezihyigitbasi) [inline presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java]: I guess this change of wrapping `load` with `try/catch` is not related to the checks you added in the constructor, but for making sure the scheduler thread doesn't die. If this is correct, it will make sense to move this change to a separate commit.

**Comment by nezihyigitbasi** [comment]
Please take a look at test failures, looks related.

File: presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java [hunk 3]
```diff
@@ -144,42 +148,49 @@ public void configure(ResourceGroup group, SelectionContext context)
         return (!globalProperties.isEmpty()) ? globalProperties.get(0).getCpuQuotaPeriod() : Optional.empty();
     }
 
-    @VisibleForTesting
-    public synchronized void load()
+    private synchronized void refresh()
```
Comment (rongrong) [inline presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/db/DbResourceGroupConfigurationManager.java]: Don't really need `synchronized` here right?

**Comment by raghavsethi** [comment]
Thanks for the reviews. I talked to @electrum and @martint and decided we don't actually want to crash the server on startup if the MySQL server is temporarily unavailable. I'll redo this and ask for another review.

**Comment by raghavsethi** [comment]
Superseded by #9567
